TestTaskHeaderKeeper.test_check_max_tasks_per_owner has a race condition

## Description

**Branch** (if launched from source): b0.19 (soon to be released 0.19.2)

**Description of the issue**:

In one build on buildbot test `TestTaskHeaderKeeper.test_check_max_tasks_per_owner` failed.

## Logs and any additional context
```
================================== FAILURES ===================================
_____________ TestTaskHeaderKeeper.test_check_max_tasks_per_owner _____________
self = <tests.golem.task.test_taskkeeper.TestTaskHeaderKeeper testMethod=test_check_max_tasks_per_owner>
    def test_check_max_tasks_per_owner(self):
    
        tk = TaskHeaderKeeper(
            environments_manager=EnvironmentsManager(),
            node=dt_p2p_factory.Node(),
            min_price=10,
            max_tasks_per_requestor=10)
        limit = tk.max_tasks_per_requestor
        new_limit = 3
    
        ids = []
        for _ in range(new_limit):
            thd = get_task_header("ta")
            ids.append(thd.task_id)
            tk.add_task_header(thd)
        last_add_time = time.time()
    
        thd = get_task_header("tb0")
        tb0_id = thd.task_id
        tk.add_task_header(thd)
    
        def _assert_headers(ids_, len_):
            ids_.append(tb0_id)
            for id_ in ids_:
                self.assertIn(id_, tk.task_headers)
            self.assertEqual(len_, len(tk.task_headers))
    
        _assert_headers(ids, len(ids) + 1)
    
        while time.time() == last_add_time:
            time.sleep(0.1)
    
        new_ids = []
        for _ in range(new_limit, limit):
            thd = get_task_header("ta")
            new_ids.append(thd.task_id)
            tk.add_task_header(thd)
    
    
        _assert_headers(ids + new_ids, limit + 1)
    
        # shouldn't remove any tasks
        tk.check_max_tasks_per_owner(thd.task_owner.key)
    
        _assert_headers(ids + new_ids, limit + 1)
    
        # Test if it skips a running task
        running_task_id = ids[0]
        tk.task_started(running_task_id)
        assert running_task_id in tk.running_tasks
        tk.max_tasks_per_requestor = tk.max_tasks_per_requestor - 1
        # shouldn't remove any tasks
        tk.check_max_tasks_per_owner(thd.task_owner.key)
    
        _assert_headers(ids + new_ids, limit + 1)
    
        # finish the task, restore state
        tk.task_ended(running_task_id)
        assert running_task_id not in tk.running_tasks
    
        tk.max_tasks_per_requestor = new_limit
    
        # should remove ta{3..9}
        tk.check_max_tasks_per_owner(thd.task_owner.key)
    
        _assert_headers(ids, new_limit + 1)
    
        # Test if it skips a running task
        running_task_id = ids[2]
        tk.task_started(running_task_id)
        assert running_task_id in tk.running_tasks
        tk.max_tasks_per_requestor = 1
        # shouldn't remove running_task_id
        tk.check_max_tasks_per_owner(thd.task_owner.key)
    
        # Should keep 0 and 2, since 2 is running
>       _assert_headers([ids[0], ids[2]], 3)
tests\golem\task\test_taskkeeper.py:370: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\golem\task\test_taskkeeper.py:318: in _assert_headers
    self.assertIn(id_, tk.task_headers)
E   AssertionError: '9f90c318-77d4-11e9-ac2c-000000007461' not found in {'9f90c319-77d4-11e9-b891-000000007461': <TaskHeader: {'mask': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'timestamp': 0, 'signature': None, 'task_id': '9f90c319-77d4-11e9-b891-000000007461', 'task_owner': {'node_name': "Bob's node", 'key': '7461', 'prv_port': None, 'pub_port': 10101, 'p2p_prv_port': None, 'p2p_pub_port': None, 'prv_addr': None, 'pub_addr': '10.10.10.10', 'prv_addresses': [], 'hyperdrive_prv_port': None, 'hyperdrive_pub_port': None, 'port_statuses': {}, 'nat_type': []}, 'deadline': 1558010271, 'subtask_timeout': 120, 'resource_size': 0, 'environment': 'DEFAULT', 'min_version': '0.19.1+dev45.gdd41c42', 'estimated_memory': 0, 'max_price': 10, 'subtasks_count': 1, 'concent_enabled': None}>, '9f90c31a-77d4-11e9-9183-000000007461': <TaskHeader: {'mask': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'timestamp': 0, 'signature': None, 'task_id': '9f90c31a-77d4-11e9-9183-000000007461', 'task_owner': {'node_name': "Bob's node", 'key': '7461', 'prv_port': None, 'pub_port': 10101, 'p2p_prv_port': None, 'p2p_pub_port': None, 'prv_addr': None, 'pub_addr': '10.10.10.10', 'prv_addresses': [], 'hyperdrive_prv_port': None, 'hyperdrive_pub_port': None, 'port_statuses': {}, 'nat_type': []}, 'deadline': 1558010271, 'subtask_timeout': 120, 'resource_size': 0, 'environment': 'DEFAULT', 'min_version': '0.19.1+dev45.gdd41c42', 'estimated_memory': 0, 'max_price': 10, 'subtasks_count': 1, 'concent_enabled': None}>, '9f90c31b-77d4-11e9-b031-000000746230': <TaskHeader: {'mask': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'timestamp': 0, 'signature': None, 'task_id': '9f90c31b-77d4-11e9-b031-000000746230', 'task_owner': {'node_name': "Bob's node", 'key': '746230', 'prv_port': None, 'pub_port': 10101, 'p2p_prv_port': None, 'p2p_pub_port': None, 'prv_addr': None, 'pub_addr': '10.10.10.10', 'prv_addresses': [], 'hyperdrive_prv_port': None, 'hyperdrive_pub_port': None, 'port_statuses': {}, 'nat_type': []}, 'deadline': 1558010271, 'subtask_timeout': 120, 'resource_size': 0, 'environment': 'DEFAULT', 'min_version': '0.19.1+dev45.gdd41c42', 'estimated_memory': 0, 'max_price': 10, 'subtasks_count': 1, 'concent_enabled': None}>}
---------------------------- Captured stderr call -----------------------------
WARNING  [golem.task.taskkeeper              ] Too many tasks, dropping 7 tasks. owner=7461..7461, ids_to_remove=['9fa173b6-77d4-11e9-979c-000000007461', '9fa173b7-77d4-11e9-9f67-000000007461', '9fa173b5-77d4-11e9-b5c6-000000007461', '9fa173b9-77d4-11e9-829e-000000007461', '9fa173b4-77d4-11e9-b98b-000000007461', '9fa173ba-77d4-11e9-9604-000000007461', '9fa173b8-77d4-11e9-bdbe-000000007461']
WARNING  [golem.task.taskkeeper              ] Too many tasks, dropping 1 tasks. owner=7461..7461, ids_to_remove=['9f90c318-77d4-11e9-ac2c-000000007461']
------------------------------ Captured log call ------------------------------
taskkeeper.py              541 WARNING  Too many tasks, dropping 7 tasks. owner=7461..7461, ids_to_remove=['9fa173b6-77d4-11e9-979c-000000007461', '9fa173b7-77d4-11e9-9f67-000000007461', '9fa173b5-77d4-11e9-b5c6-000000007461', '9fa173b9-77d4-11e9-829e-000000007461', '9fa173b4-77d4-11e9-b98b-000000007461', '9fa173ba-77d4-11e9-9604-000000007461', '9fa173b8-77d4-11e9-bdbe-000000007461']
taskkeeper.py              541 WARNING  Too many tasks, dropping 1 tasks. owner=7461..7461, ids_to_remove=['9f90c318-77d4-11e9-ac2c-000000007461']
```

--- a/tests/golem/task/test_taskkeeper.py
+++ b/tests/golem/task/test_taskkeeper.py
@@ -284,17 +284,25 @@ class TestTaskHeaderKeeper(LogTestCase):
             max_tasks_per_requestor=10)
         limit = tk.max_tasks_per_requestor
         new_limit = 3
+        last_add_time = time.time()
 
         ids = []
         for _ in range(new_limit):
             thd = get_task_header("ta")
             ids.append(thd.task_id)
             tk.add_task_header(thd)
-        last_add_time = time.time()
+
+            while time.time() == last_add_time:
+                time.sleep(0.1)
+            last_add_time = time.time()
 
         thd = get_task_header("tb0")
         tb0_id = thd.task_id
         tk.add_task_header(thd)
+
+        while time.time() == last_add_time:
+            time.sleep(0.1)
+        last_add_time = time.time()
 
         def _assert_headers(ids_, len_):
             ids_.append(tb0_id)
@@ -304,14 +312,15 @@ class TestTaskHeaderKeeper(LogTestCase):
 
         _assert_headers(ids, len(ids) + 1)
 
-        while time.time() == last_add_time:
-            time.sleep(0.1)
-
         new_ids = []
         for _ in range(new_limit, limit):
             thd = get_task_header("ta")
             new_ids.append(thd.task_id)
             tk.add_task_header(thd)
+
+            while time.time() == last_add_time:
+                time.sleep(0.1)
+            last_add_time = time.time()
 
 
         _assert_headers(ids + new_ids, limit + 1)

--- a/tests/golem/task/test_taskkeeper.py
+++ b/tests/golem/task/test_taskkeeper.py
@@ -234,7 +234,6 @@ class TestTaskHeaderKeeper(LogTestCase):
             thd = get_task_header("ta")
             ids.append(thd.task_id)
             tk.add_task_header(thd)
-        last_add_time = time.time()
 
         for id_ in ids:
             self.assertIn(id_, tk.task_headers)
@@ -248,9 +247,7 @@ class TestTaskHeaderKeeper(LogTestCase):
 
         self.assertIn(tb_id, tk.task_headers)
 
-        while time.time() == last_add_time:
-            frozen_time.tick(  # pylint: disable=no-member
-                delta=timedelta(milliseconds=100))
+        frozen_time.tick(timedelta(seconds=0.1))  # pylint: disable=no-member
 
         thd = get_task_header("ta")
         new_task_id = thd.task_id
@@ -261,8 +258,7 @@ class TestTaskHeaderKeeper(LogTestCase):
             self.assertIn(id_, tk.task_headers)
         self.assertIn(tb_id, tk.task_headers)
 
-        frozen_time.tick(  # pylint: disable=no-member
-            delta=timedelta(milliseconds=100))
+        frozen_time.tick(timedelta(seconds=0.1))  # pylint: disable=no-member
         tk.remove_old_tasks()
 
         thd = get_task_header("ta")
@@ -275,7 +271,9 @@ class TestTaskHeaderKeeper(LogTestCase):
             self.assertIn(ids[i], tk.task_headers)
         self.assertIn(tb_id, tk.task_headers)
 
-    def test_check_max_tasks_per_owner(self):
+    @freeze_time(as_arg=True)
+    # pylint: disable=no-self-argument
+    def test_check_max_tasks_per_owner(freezer, self):
 
         tk = TaskHeaderKeeper(
             environments_manager=EnvironmentsManager(),
@@ -284,7 +282,6 @@ class TestTaskHeaderKeeper(LogTestCase):
             max_tasks_per_requestor=10)
         limit = tk.max_tasks_per_requestor
         new_limit = 3
-        last_add_time = time.time()
 
         ids = []
         for _ in range(new_limit):
@@ -292,17 +289,13 @@ class TestTaskHeaderKeeper(LogTestCase):
             ids.append(thd.task_id)
             tk.add_task_header(thd)
 
-            while time.time() == last_add_time:
-                time.sleep(0.1)
-            last_add_time = time.time()
+            freezer.tick(timedelta(seconds=0.1))  # pylint: disable=no-member
 
         thd = get_task_header("tb0")
         tb0_id = thd.task_id
         tk.add_task_header(thd)
 
-        while time.time() == last_add_time:
-            time.sleep(0.1)
-        last_add_time = time.time()
+        freezer.tick(timedelta(seconds=0.1))  # pylint: disable=no-member
 
         def _assert_headers(ids_, len_):
             ids_.append(tb0_id)
@@ -318,10 +311,7 @@ class TestTaskHeaderKeeper(LogTestCase):
             new_ids.append(thd.task_id)
             tk.add_task_header(thd)
 
-            while time.time() == last_add_time:
-                time.sleep(0.1)
-            last_add_time = time.time()
-
+            freezer.tick(timedelta(seconds=0.1))  # pylint: disable=no-member
 
         _assert_headers(ids + new_ids, limit + 1)
 
@@ -472,7 +462,7 @@ class TestCompTaskKeeper(LogTestCase, PEP8MixIn, TempDirFixture):
 
         test_headers = []
         test_subtasks_ids = []
-        for x in range(10):
+        for _ in range(10):
             header = get_task_header()
             header.deadline = timeout_to_deadline(1)
             header.subtask_timeout = 3

